src/cell/main.c

```diff
@@ -86,6 +89,124 @@ main(int argc, char* argv[])
 	return retval;
 }
 
+static bool
+generate_project(char* path)
+{
+	char* project_name;
+	char* author;
+	char* width;
+	char* height;
+	char* summary;
+	path_t* app_path;
+	path_t* project_path;
+	path_t* template_path;
+	path_t* tmpl_path;
+	path_t* mjs_path;
+	FILE* mjs_file = NULL;
+	char* tmpl_str;
+	char* mjs_str;
+	bool success = false;
+
+	printf("Enter game name: ");
+	project_name = malloc(256);
+	fgets(project_name, 256, stdin);
+	project_name[strcspn(project_name, "\r\n")] = 0;
+
+	printf("Enter author: ");
+	author = malloc(256);
+	fgets(author, 256, stdin);
+	author[strcspn(author, "\r\n")] = 0;
+
+	printf("Enter screen width: ");
+	width = malloc(256);
+	fgets(width, 256, stdin);
+	width[strcspn(width, "\r\n")] = 0;
+
+	printf("Enter screen height: ");
+	height = malloc(256);
+	fgets(height, 256, stdin);
+	height[strcspn(height, "\r\n")] = 0;
+
+	printf("Enter game summary: ");
+	summary = malloc(256);
+	fgets(summary, 256, stdin);
+	summary[strcspn(summary, "\r\n")] = 0;
+
+	project_path = path_new_dir(path);
+	path_mkdir(project_path);
+
+	app_path = path_new_self();
+	template_path = path_rebase(path_new("template/"), app_path);
+	if (!path_resolve(template_path, NULL)) {
+		path_free(template_path);
+		template_path = path_rebase(path_new("../share/minisphere/template"), app_path);
+		path_resolve(template_path, NULL);
```

Comment:
I feel like this whole path resolution bit could be simpler but I struggle to imagine how right now.  Something to deal with post-merge I guess.

Reply:
fs.c does almost exactly the same thing when it creates the system directory. Since both system/ and template/ are placed in the same directory, I saw no reason to change it. I don't have a mac so I don't know if "/usr/bin/../share" would cause undefined behavior.

Reply:
I’d have to ask @rhuanjl about this; I do know the Mac builds are bundled and have a totally different structure to the usual Unix conventions.  Most likely the template directory would go into the Resources directory of the bundle and picked up by the same code used to find it on Windows.

Reply:
Use of /usr for anything on macOS is deprecated/advised against - it's basically for devtools only.

Where exactly are we placing this templates folder? If it's just for cell I'd probably go with putting it in the folder that cell is in which should work fine (note cell is not in a bundle as it's a command line tool).

Reply:
In Windows installations the template directory is placed in the main directory, alongside the engine and devtools:
https://github.com/fatcerberus/minisphere/blob/master/setup/minisphere.iss#L108

Therefore placing it alongside the `cell` executable should work without any code changes... I think.